src/NzbDrone.Core/Notifications/Trakt/TraktService.cs

```diff
@@ -0,0 +1,269 @@
+                    return new ValidationFailure("Token", "Access Token is invalid");
+                }
+
+                _logger.Error(ex, "Unable to send test message: " + ex.Message);
+                return new ValidationFailure("Token", "Unable to send test message");
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, "Unable to send test message: " + ex.Message);
+                return new ValidationFailure("", "Unable to send test message");
+            }
+        }
+
+        public void AddEpisodeToCollection(TraktSettings settings, Series series, EpisodeFile episodeFile)
+        {
+            var payload = new TraktCollectShowsResource
+            {
+                Shows = new List<TraktCollectShow>()
+            };
+
+            var traktResolution = MapResolution(episodeFile.Quality.Quality.Resolution, episodeFile.MediaInfo?.ScanType);
+            var mediaType = MapMediaType(episodeFile.Quality.Quality.Source);
+            var audio = MapAudio(episodeFile);
+            var audioChannels = MapAudioChannels(episodeFile, audio);
+
+            var payloadEpisodes = new List<TraktEpisodeResource>();
+
+            foreach (var episode in episodeFile.Episodes.Value)
+            {
+                payloadEpisodes.Add(new TraktEpisodeResource
+                {
+                    Number = episode.EpisodeNumber,
+                    CollectedAt = DateTime.Now,
+                    Resolution = traktResolution,
+                    MediaType = mediaType,
+                    AudioChannels = audioChannels,
+                    Audio = audio,
+                });
+            }
+            
+            var payloadSeasons = new List<TraktSeasonResource>();
+            payloadSeasons.Add(new TraktSeasonResource
+            {
+                Number = episodeFile.SeasonNumber,
+                Episodes = payloadEpisodes
+            });
+
+            payload.Shows.Add(new TraktCollectShow
+            {
+                Title = series.Title,
+                Year = series.Year,
+                Ids = new TraktShowIdsResource
+                {
+                    Tvdb = series.TvdbId,
+                    Imdb = series.ImdbId ?? "",
+                },
+                Seasons = payloadSeasons,
+            }); ;
+
+            _proxy.AddToCollection(payload, settings.AccessToken);
+        }
+
+        private string MapMediaType(QualitySource source)
+        {
+            var traktSource = string.Empty;
+
+            switch (source)
+            {
+                case QualitySource.Web:
+                case QualitySource.WebRip:
+                    traktSource = "digital";
+                    break;
+                case QualitySource.BlurayRaw:
+                case QualitySource.Bluray:
+                    traktSource = "bluray";
+                    break;
+                case QualitySource.Television:
```

Comment:
I think we should put TV in the `vhs` media type, it looks to be the closest option and correctly separates it from DVD.

Reply:
I agree, that it is good to differentiate DVD from other TV recordings, but VHS isn't entirely accurate either.

@rg9400 tried to contact Trakt Dev to inquire about adding more options. We aren't too sure if they will respond. IN the meantime I will change to VHS as per @markus101 's suggestion. It is easy enough to change in the future if better options arise.

Reply:
https://support.trakt.tv/forums/188762-general/suggestions/41164723-more-collection-metadata-options

Reply:
Looks like it's been completed, but I don't know what exactly they added as their API docs don't seem to have been updated (the 'MEDIA OBJECT POST DATA' section).

Reply:
It seems that Trakt folks are still debating whether or not they will go ahead with the TV option.
Thanks to @rg9400 for looking into this with the trakt Folks: https://forums.trakt.tv/t/more-metadata-options-for-collections/711/12